### PR TITLE
CE-1771 Flags rendering

### PR DESCRIPTION
**Pull request by adamkarminski**
This pull request deals with rendering of flags. It injects wikitext to a Parser's `$text` using the `ParserBeforeInternalParse` hook. 

It shouldn't fire for edit pages and VisualEditor API calls.

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/Flags/Flags.hooks.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+		/**
+		 * Don't check for flags if:
+		 * - you've already checked
+		 * - a user is on an edit page
+		 * - the request is from VE
+		 */
+		if ( !$parser->mFlagsParsed
+			&& !\WikiaPageType::isEditPage()
+			&& !( $wgRequest->getVal( 'action' ) == 'visualeditor' )
```
Comment (lgarczewski) [inline extensions/wikia/Flags/Flags.hooks.php]: Huh? Where are we using this value for action? Isn't it veaction=edit ?

Comment (adamkarminski) [inline extensions/wikia/Flags/Flags.hooks.php]: The "internal" API calls from VE don't have the `veaction` parameter but `action=visualeditor` and `paction=parse`.

**Comment by lgarczewski** [comment]
One comment, otherwise looks good.

File: extensions/wikia/Flags/Flags.setup.php [hunk 2]
```diff
@@ -26,3 +26,24 @@
...
+
+/**
+ * Views
+ */
+$wgAutoloadClasses['Flags\Views\FlagView'] = __DIR__ . '/views/FlagView.class.php';
+
+/**
+ * Helper
+ */
+$wgAutoloadClasses['FlagsHelper'] = __DIR__ . '/FlagsHelper.class.php';
```
Comment (Grunny) [inline extensions/wikia/Flags/Flags.setup.php]: Any reason the helper isn't in the Flags namespace too?

Comment (adamkarminski) [inline extensions/wikia/Flags/Flags.setup.php]: Haha, you've been faster by a minute! ;)

**Comment by lgarczewski** [comment]
LGTM after changes made.

**Pull request merged**